internal/backends/android-activity/javahelper.rs

```diff
@@ -384,20 +410,20 @@ extern "system" fn Java_SlintAndroidJavaHelper_updateText(
     .unwrap()
 }
 
-fn convert_utf16_index_to_utf8(in_str: &str, utf16_index: usize) -> usize {
-    let mut utf16_counter = 0;
+fn convert_char_index_to_utf8(in_str: &str, char_index: usize) -> usize {
+    let mut char_counter = 0;
 
-    for (utf8_index, c) in in_str.char_indices() {
-        if utf16_counter >= utf16_index {
+    for (utf8_index, _) in in_str.char_indices() {
```

Comment:
I think this could be implemented with `in_str.char_indices().skip(char_index).0` or something like that.

But are you sure UTF-32 is the right encoding in Java? I was under the impression it was UTF-16

Reply:
Sorry, I made a mistake. I recovered it according to your original code.